go/multipooler/manager/rpc_consensus.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2025 Supabase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	clustermetadatapb "github.com/multigres/multigres/go/pb/clustermetadata"
+	consensusdatapb "github.com/multigres/multigres/go/pb/consensusdata"
+	pgctldpb "github.com/multigres/multigres/go/pb/pgctldservice"
+)
+
+// BeginTerm handles coordinator requests during leader appointments
+func (pm *MultiPoolerManager) BeginTerm(ctx context.Context, req *consensusdatapb.BeginTermRequest) (*consensusdatapb.BeginTermResponse, error) {
+	// CRITICAL: Must be able to reach Postgres to participate in cohort
+	if pm.db == nil {
+		return nil, fmt.Errorf("postgres unreachable, cannot accept new term")
+	}
+
+	// Test database connectivity
+	if err := pm.db.PingContext(ctx); err != nil {
+		return nil, fmt.Errorf("postgres unhealthy, cannot accept new term: %w", err)
+	}
+
+	// Load consensus term from disk if not already loaded
+	pm.mu.Lock()
+	if pm.consensusTerm == nil {
```

Comment:
Related to your comment below, for follow up. Probably we want to encapsulate all this is a convenience function